{
    "status": "Opinion", 
    "last_updated": "2013-03-08 19:04:35.281542+00:00", 
    "description": "The CA code doesn't get install when running \"python setup.py install\". This renders an install of nova useless by default.\n\nWe can either fix up the CA code to let it live in the python path where the templates also (soon will) live and use $state_path/CA for its state, or we can move it all into Python.", 
    "tags": [], 
    "importance": "Wishlist", 
    "heat": 24, 
    "link": "https://bugs.launchpad.net/nova/+bug/683137", 
    "owner": "None", 
    "id": 683137, 
    "index": 2243, 
    "openned": "2010-11-30 14:25:51.379683+00:00", 
    "created": "2010-11-30 14:25:51.379683+00:00", 
    "title": "CA code should be rewritten in Python", 
    "comments": [
        {
            "content": "The CA code doesn't get install when running \"python setup.py install\". This renders an install of nova useless by default.\n\nWe can either fix up the CA code to let it live in the python path where the templates also (soon will) live and use $state_path/CA for its state, or we can move it all into Python.", 
            "date_created": "2010-11-30 14:25:51.379683+00:00", 
            "author": "https://api.launchpad.net/1.0/~soren"
        }, 
        {
            "content": "Soren is working on the long-term fix, but we could use a short-term one. Unassigning so that someone else can work on it.\r\n\r\nWorkaround:\r\n<soren> georgy_: As a temporary measure, you could probably just copy the CA direcotry from the tarball into /usr/local/lib/python2.6/dist-packages/nova-2011.1-py2.6.egg/\r\n<georgy_> that worked thanks", 
            "date_created": "2011-01-24 10:58:34.256695+00:00", 
            "author": "https://api.launchpad.net/1.0/~ttx"
        }, 
        {
            "content": "Splitting bug in two. One is the rewrite CA code in Python (this bug), the other is the fix for current python setup.py install (bug 727794).", 
            "date_created": "2011-03-17 11:07:06.200370+00:00", 
            "author": "https://api.launchpad.net/1.0/~ttx"
        }, 
        {
            "content": "Can somebody revisit this bug and determine its validity?", 
            "date_created": "2012-01-29 21:00:58.372258+00:00", 
            "author": "https://api.launchpad.net/1.0/~bcwaldon"
        }, 
        {
            "content": "The code in nova/crypto.py still calls out to the commandline for openssl, so I assume this bug is still valid.", 
            "date_created": "2012-05-16 11:32:39.722430+00:00", 
            "author": "https://api.launchpad.net/1.0/~fifieldt"
        }, 
        {
            "content": "it is still valid, but I personally think it is fine to shell out.  The python crypto libraries are a bit annoying.", 
            "date_created": "2012-05-16 17:45:27.267023+00:00", 
            "author": "https://api.launchpad.net/1.0/~vishvananda"
        }, 
        {
            "content": "moving to opinion. It's fine if someone wants to resurect this later, but we don't need to carry it in the tracker.", 
            "date_created": "2013-03-08 19:04:34.622653+00:00", 
            "author": "https://api.launchpad.net/1.0/~sdague"
        }
    ], 
    "closed": "2013-03-08 19:04:07.422089+00:00"
}